app/javascript/components/pages/Home.js

```diff
@@ -4,22 +4,54 @@ import { Nav, NavItem } from 'reactstrap'
 export default class Home extends Component {
   render() {
     const {
+      logged_in,
+      current_user,
       new_user_route,
       sign_in_route,
+      sign_out_route
     } = this.props
+    console.log("logged_in:", logged_in)
+    console.log("current_user:", current_user)
     return (
       <div>
-        <h3>Welcome!</h3>
+       {!logged_in && <h3>Welcome!</h3>}
+       {!logged_in && (
```

Comment:
There is a lot of repeated code between logged out and logged in.  Information that will be displayed on both can be outside the conditional rendering.